src/java_bytecode/java_bytecode_convert_class.cpp

```diff
@@ -45,13 +45,14 @@ class java_bytecode_convert_classt:public messaget
   {
     add_array_types();
 
-    if(parse_tree.loading_successful)
+    bool loading_success=parse_tree.loading_successful;
+    if(loading_success)
       convert(parse_tree.parsed_class);
-    else if(string_preprocess.is_known_string_type(
-              parse_tree.parsed_class.name))
+
```

Comment:
Yes we now want to replace string types even if they are loaded.
The idea behind this is that we will provide through the `models-library` a String class which will be loaded in the classpath, but we still want to use the type we define internally for strings.
The class still needs to be converted though, because we want to use some methods that are defined in the String.class provided.